protocol/kafka_confluent/v2/option.go

```diff
@@ -0,0 +1,178 @@
+		if topics == nil {
+			return fmt.Errorf("the consumer topics option must not be nil")
+		}
+		p.consumerTopics = topics
+		return nil
+	}
+}
+
+// WithRebalanceCallBack sets the callback for rebalancing of the consumer group. This option is not required.
+func WithRebalanceCallBack(rebalanceCb kafka.RebalanceCb) Option {
+	return func(p *Protocol) error {
+		if rebalanceCb == nil {
+			return fmt.Errorf("the consumer group rebalance callback must not be nil")
+		}
+		p.consumerRebalanceCb = rebalanceCb
+		return nil
+	}
+}
+
+// WithPollTimeout sets timeout of the consumer polling for message or events, return nil on timeout. This option is not required.
+func WithPollTimeout(timeoutMs int) Option {
+	return func(p *Protocol) error {
+		p.consumerPollTimeout = timeoutMs
+		return nil
+	}
+}
+
+// WithSender set a kafka.Producer instance to init the client directly. This option is not required.
+func WithSender(producer *kafka.Producer) Option {
+	return func(p *Protocol) error {
+		if producer == nil {
+			return fmt.Errorf("the producer option must not be nil")
+		}
+		p.producer = producer
+		return nil
+	}
+}
+
+// WithSender set a kafka.Consumer instance to init the client directly. This option is not required.
+func WithReceiver(consumer *kafka.Consumer) Option {
+	return func(p *Protocol) error {
+		if consumer == nil {
+			return fmt.Errorf("the consumer option must not be nil")
+		}
+		p.consumer = consumer
+		return nil
+	}
+}
+
+// Opaque key type used to store topicPartitionOffsets: assign them from ctx. This option is not required.
+type topicPartitionOffsetsType struct{}
+
+var offsetKey = topicPartitionOffsetsType{}
+
+// WithTopicPartitionOffsets will set the positions where the consumer starts consuming from. This option is not required.
+func WithTopicPartitionOffsets(ctx context.Context, topicPartitionOffsets []kafka.TopicPartition) context.Context {
+	if len(topicPartitionOffsets) == 0 {
+		panic("the topicPartitionOffsets cannot be empty")
+	}
+	for _, offset := range topicPartitionOffsets {
+		if offset.Topic == nil || *(offset.Topic) == "" {
+			panic("the kafka topic cannot be nil or empty")
+		}
+		if offset.Partition < 0 || offset.Offset < 0 {
+			panic("the kafka partition/offset must be non-negative")
+		}
+	}
+	return context.WithValue(ctx, offsetKey, topicPartitionOffsets)
+}
+
+// TopicPartitionOffsetsFrom looks in the given context and returns `[]kafka.TopicPartition`
+func TopicPartitionOffsetsFrom(ctx context.Context) []kafka.TopicPartition {
+	c := ctx.Value(offsetKey)
+	if c != nil {
+		if s, ok := c.([]kafka.TopicPartition); ok {
+			return s
+		}
+	}
+	return nil
+}
+
+// Opaque key type used to store topic partition
+type topicPartitionKeyType struct{}
+
+var topicPartitionKey = topicPartitionKeyType{}
+
+// WithTopicPartition returns back a new context with the given partition. This option is not required.
+func WithTopicPartition(ctx context.Context, partition int32) context.Context {
```

Comment:
Wondering where/how this would be used? For example, there's also `WithTopicPartitionsOffsets` which I found similar and the latter one referencing a topic whereas this function just takes a partition (for which topic)?